test/acceptance/features/steps/steps.py

```diff
@@ -145,26 +152,35 @@ def db_instance_is_running(context, db_name):
 @when(sbr_is_applied_step)
 def sbr_is_applied(context):
     sbr_yaml = context.text
-    sbr = ServiceBindingRequest()
-    if context.__contains__("nodejs_app"):
-        application = context.nodejs_app
-        context.nodejs_app_original_generation = application.get_observed_generation()
-        context.nodejs_app_original_pod_name = application.get_running_pod_name()
+    namespace = context.namespace
+    sbr = ServiceBindingRequest(namespace.name)
+    if context.__contains__("application"):
+        application = context.application
+        context.application_original_generation = application.get_observed_generation()
+        if context.application_type.__contains__("knative"):
+            context.application_original_rev_name = application.get_rev_name()
+        else:
+            context.application_original_pod_name = application.get_running_pod_name()
     sbr.create(sbr_yaml) | should.be_truthy.desc("Service Binding Request Created")
 
-
 # STEP
+
+
 @then(u'application should be re-deployed')
 def then_application_redeployed(context):
-    application = context.nodejs_app
-    application.get_redeployed_pod_name(context.nodejs_app_original_pod_name) | should_not.be_none.desc(
-        "There is a running pod of the application different from the original one before redeployment.")
+    application = context.application
+    if context.application_type.__contains__("knative"):
+        application.get_redeployed_rev_name(context.application_original_rev_name) | should_not.be_none.desc(
+            "There is a running revision of the application different from the original one before redeployment")
+    else:
+        application.get_redeployed_pod_name(context.application_original_pod_name) | should_not.be_none.desc(
+            "There is a running pod of the application different from the original one before redeployment.")
```

Comment:
we should really avoid this if/then/switch statements... instead create a common api... if we want to know if an application  could redeployed, then we could have a method e.g. `is_redeployed()` which would return `true/false`. `context.application_original_rev_name` or `context.application_original_pod_name` already known internal application properties, so they do not need to preserved explicitly in context.

Reply:
sure, this has changed to this, please confirm. Now there are no checks for application type as knative or nodejs
`@then(u'application should be re-deployed')
def then_application_redeployed(context):
    application = context.application
    assert application.is_redeployed(context.application_original_generation) is not None, "application is not redeployed"`